lightning/src/ln/channel.rs

```diff
@@ -5687,8 +5730,25 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 			(9, target_closing_feerate_sats_per_kw, option),
 			(11, monitor_pending_finalized_fulfills, vec_type),
 			(13, channel_creation_height, option),
+			(15, preimages_opt, vec_type),
 		});
 
+		if let Some(preimages) = preimages_opt {
+			let mut iter = preimages.into_iter();
+			for htlc in pending_outbound_htlcs.iter_mut() {
+				match &htlc.state {
+					OutboundHTLCState::AwaitingRemoteRevokeToRemove(OutboundHTLCOutcome::Success(None)) => {
+						htlc.state = OutboundHTLCState::AwaitingRemoteRevokeToRemove(OutboundHTLCOutcome::Success(iter.next().expect("not enough preimages")));
```

Comment:
Shouldn't `expect` in deser, we should return `InvalidValue` instead.

Reply:
fixed